src/codegen/compile.ml

```diff
@@ -532,8 +532,17 @@ module E = struct
     | ts -> VarBlockType (nr (func_type env (FuncType ([], ts))))
 
   let if_ env tys thn els = G.if_ (as_block_type env tys) thn els
+
+  (* NB: confuses wasm-opt, don't use for now
+  let _multi_if_ env tys1 tys2 thn els =
```

Comment:
Just out of curiosity: Is `wasm-opt` failing or creating invalid code?

Reply:
I don't fully recall, but I think it was failing.

Reply:
Commit https://github.com/dfinity/motoko/pull/4171/commits/280ed54d50d5fb8020ef2f85aa05ca358d5f42ef produces this error. Basically I was trying to conditionally call one of two functions with args already on the stack and wasm-opt barfed:

```
test-bench> @@ -0,0 +1,4 @@
test-bench> +Error: Failed to read module
test-bench> +
test-bench> +Caused by:
test-bench> +    [parse exception: Block requires more values than are available (at 0:57416)]

Reply:
I should probably be a good citizen and report the bug, but I'd rather make progress..